oauth2client/devshell.py

```diff
@@ -117,6 +124,10 @@ def __init__(self, user_agent=None):
     def _refresh(self, http_request):
         self.devshell_response = _SendRecv()
         self.access_token = self.devshell_response.access_token
+        expires_in = self.devshell_response.expires_in
+        if expires_in is not None:
```

Comment:
As currently implemented, self.devshell_response.expires_in will not be none as we always fill in that value.  However, the value is marked as optional in the proto definition, so in theory yes it could be.

self.token_expiry can be non None if refresh() was previously called and set for the last value of the access token.  However, in the case where we get the same access token back, the expires_in value will also be less by whatever time delta has passed, so it will set it to the same token_expiry.